pkg/distributor/distributor.go

```diff
@@ -167,6 +170,32 @@ type Config struct {
 	OTLPConfig OTLPConfig `yaml:"otlp"`
 }
 
+type PushHandlerMetrics struct {
+	pushRequestSizeBytes *prometheus.HistogramVec
+}
+
+func NewPushHandlerMetrics(reg prometheus.Registerer) *PushHandlerMetrics {
+	return &PushHandlerMetrics{
+		pushRequestSizeBytes: promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+			Name:                            "cortex_distributor_push_requests_uncompressed_size_bytes",
+			Help:                            "Histogram of push request's uncompressed size in bytes",
+			NativeHistogramBucketFactor:     1.1,
+			NativeHistogramMinResetDuration: 1 * time.Hour,
+			NativeHistogramMaxBucketNumber:  100,
+		}, []string{"user", "format"}),
```

Comment:
I wonder if we really need the `format` parameter here. Why would we need different limits for different formats? Do we plan to have a new limit for RW 2.0?

I feel we should consolidate the request size limit into one https://github.com/cortexproject/cortex/pull/6333.

Reply:
I have no plan to something limit for PRW 2.0. It could be that we can add some limits. But, I thought different formats (prw1, prw2, and otlp) would have different body sizes.

So, I attached the `format` label so that the user can configure limit configs for formats.

Reply:
If we want to track different formats then for the same reason we need to differentiate compression formats as well as they have different sizes. OTLP has protobuf and JSON then we also need to differentiate that.

To simplify things I think it is good enough to use a single size.

Reply:
Ok, then detach `format` and  the`cortex_distributor_push_requests_uncompressed_size_bytes` tracks the uncompressed size of all of the formats.
Next step is to delete `distributor.otlp-max-recv-msg-size` flags.

Reply:
Let's hear from other maintainers...
@friedrichg @danielblando @alanprot @CharlieTLe ?